Added version 9.12.0

bahugo opened the pull request.

bahugo committed:
Added version 9.12.0

bahugo committed:
Added Requires-Python to METADATA.in

bahugo committed:
Fixed matix in github action

bahugo commented:
closes #1

jschueller commented on .github/workflows/stable.yml:
```diff
@@ -7,7 +7,7 @@ jobs:
     strategy:
       matrix:
-        version: [9.11.0]
+        version: [9.11.0, 9.12.0]
```
lets drop 9.11

bahugo replied:
I've found another issue #3. It might be interesting to fix also 9.11.

bahugo replied:
Do you want me to open another pull request for #3 and then remove version 9.11 from this one?

jschueller replied:
lets do everything in one PR

jschueller commented on .github/workflows/stable.yml:
```diff
@@ -37,22 +37,19 @@ jobs:
         include:
           - abi: cp38
             python-version: '3.8'
-            version: 9.11.0
```
I think this is still needed (update it)

bahugo replied:
I tkink it should work https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations but i haven't tested it.

bahugo committed:
Adds write_distinfo python script

bahugo committed:
Updated med-file download link

bahugo committed:
Corrected salome link

bahugo committed:
Updated med-file download link

bahugo committed:
Updated link

bahugo committed:
updated unzip

bahugo committed:
Modified extraction on linux

bahugo committed:
Found direct link for med

bahugo committed:
Modified extraction

bahugo committed:
Fixed extraction on linux

bahugo committed:
Fixed folder name in .bat compilation

bahugo committed:
Updated github action version

bahugo committed:
revert med-file version increment

bahugo committed:
Fixed extraction for 4.1.1 med tar.gz

bahugo committed:
Added __version__

bahugo committed:
Fixed write_distinfo.py call

bahugo committed:
changed write_distinfo call

bahugo committed:
Fixed METADATA.in path

bahugo committed:
Fixed metadata_file path in write_distinfo.py

bahugo committed:
Added check in write_distinfo

bahugo committed:
Using pyproject.toml and python build to build wheel (WIP)

bahugo committed:
Using pyproject.toml and python build to build wheel (WIP)

bahugo committed:
Changed wheel generation path

bahugo committed:
(WIP) added print in .bat

bahugo committed:
(WIP) Reworking windows compilation

bahugo committed:
(WIP) windows installation

bahugo committed:
(WIP) added artifacts

bahugo committed:
(WIP) added artifacts

bahugo committed:
(WIP) compilation on windows

bahugo committed:
(WIP) added echo

bahugo committed:
(WIP) windows compilation

bahugo committed:
(WIP) windows compilation

bahugo committed:
(WIP) windows compilation

bahugo committed:
(WIP) compilation windows

bahugo committed:
(WIP) windows compilation

bahugo committed:
(WIP) compilation windows

bahugo committed:
(WIP) windows compilation

bahugo commented:
Hi, 
I've found the new download link for med-4.1.1.
I've added a pyproject.toml to simplify the built of the wheel.
But I'm still not able to compile medcoupling...
I've been focusing my efforts on windows.
I've reworked a bit the .bat file in order to be able de test it on my local windows machine.

Here are the failing compilations : 
- metis
- med
- medcoupling

I've tried (a lot) to figure out by my own what's the issue, i've event tried to use your legacy .bat (and just change the med download link) but it failed too.

Any help would be more than welcome 😉.
